packages/react-reconciler/src/ReactFiberCompleteWork.js

```diff
@@ -524,6 +517,18 @@ function completeWork(
           'React. Please file an issue.',
       );
   }
+
+  if (enableProfilerTimer) {
+    if (workInProgress.mode & ProfileMode) {
+      // Don't record elapsed time unless the "complete" phase has succeeded.
+      // Certain renderers may error during this phase (i.e. ReactNative View/Text nesting validation).
+      // If an error occurs, we'll mark the time while unwinding.
+      // This simplifies the unwinding logic and ensures consistency.
+      recordElapsedActualRenderTime(workInProgress);
```

Comment:
Moving this to the end of `completeWork` was necessary to avoid double-popping in the event of an error (like RN's `Text`/`View` validation).